### PR TITLE
[lg-2] Add user and requestId logging

### PR DESCRIPTION
**Pull request by groksana**

### CODE REVIEW

File: movieland-service-api/src/main/java/com/gromoks/movieland/service/security/AuthorizationService.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.gromoks.movieland.service.security;
 
-import com.gromoks.movieland.service.entity.UserToken;
+import com.gromoks.movieland.entity.User;
 
 public interface AuthorizationService {
-    void authorizeToAddReview(UserToken userToken);
+    void authorizeToAddReview(User user);
```
Comment (TakeALookHere) [inline movieland-service-api/src/main/java/com/gromoks/movieland/service/security/AuthorizationService.java]: name of method. You'll authorize user not only for review adding

File: movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        return true;
+    }
+
+    @Override
+    public void afterCompletion(HttpServletRequest request, HttpServletResponse response,Object handler, Exception e) throws Exception {
+        MDC.clear();
+    }
+
+   /* private String authenticate(String uuid) throws AuthenticationException {
```
Comment (TakeALookHere) [inline movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java]: commented code

File: movieland-service-impl/src/main/java/com/gromoks/movieland/service/impl/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -17,6 +18,8 @@
 
     private final Logger log = LoggerFactory.getLogger(getClass());
 
+    private static final ThreadLocal<User> threadLocalScope = new  ThreadLocal<>();
```
Comment (Jokeroff) [inline movieland-service-impl/src/main/java/com/gromoks/movieland/service/impl/security/AuthenticationServiceImpl.java]: uppercase

File: movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.util.UUID;
+
+public class LoggerInterceptor extends HandlerInterceptorAdapter {
+
+    private static Logger log = LoggerFactory.getLogger(LoggerInterceptor.class);
+    private static final String GUEST = "guest";
+
+    @Autowired
+    private AuthenticationService authenticationService;
```
Comment (Jokeroff) [inline movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java:20]: not required - you use static methods

File: movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java (hunk 2 above)
Comment (Jokeroff) [inline movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java]: commented code (

File: movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/SecurityInterceptor.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        return true;
+    }
+
+    private void authenticate(String uuid) throws AuthenticationException {
+        if (uuid != null) {
+            UserToken userToken;
+            try {
+                userToken = authenticationService.getAuthenticationByUuid(uuid);
+            } catch (AuthenticationException e) {
+                log.error("Expired or invalid uuid");
```
Comment (Jokeroff) [inline movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/SecurityInterceptor.java]: adding uuid in log will be useful

File: movieland-persistence-jdbc/src/main/java/com/gromoks/movieland/dao/jdbc/JdbcReviewDao.java [hunk 6]
```diff
@@ -20,14 +22,20 @@
...
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
         MapSqlParameterSource parameterSource = new MapSqlParameterSource();
         parameterSource.addValue("movieId",review.getMovieId());
         parameterSource.addValue("userId",review.getUser().getId());
         parameterSource.addValue("text",review.getText());
-        namedParameterJdbcTemplate.update(addReviewSQL,parameterSource);
+        namedParameterJdbcTemplate.update(addReviewSQL,parameterSource,keyHolder, new String[]{"id"});
+
+        review.setId(keyHolder.getKey().intValue());
```
Comment (DanwerOK) [inline movieland-persistence-jdbc/src/main/java/com/gromoks/movieland/dao/jdbc/JdbcReviewDao.java]: :+1:

File: movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/SecurityInterceptor.java (hunk 5 above)
Comment (groksana) [reply movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/SecurityInterceptor.java]: It's confident )

File: movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java (hunk 4 above)
Comment (groksana) [reply movieland-web/src/main/java/com/gromoks/movieland/web/interceptor/LoggerInterceptor.java:20]: It's not static methods but make sense to do static

**Review by Jokeroff — COMMENTED** [review]
reviewed

**Pull request merged**